src/working/bst.py

```diff
@@ -0,0 +1,143 @@
+"""Python implementation of Binary Search Tree."""
+from timeit import timeit
+
+
+class BST(object):
+    """Binary Search Tree."""
+
+    def __init__(self, iterable=None):
+        """Initialize Binary Search tree."""
+        self._root = None
+        self._length = 0
+        self._rdepth = 0
+        self._ldepth = 0
+        self._depth = 0
+        self._balance = 0
+        if type(iterable) in [tuple, list]:
+            for i in iterable:
+                if type(i) in [int, float]:
+                    self.insert(i)
+                else:
+                    raise TypeError('''
+Try again with only numbers in your list or tuple.''')
+        elif type(iterable) == int:
+            self.insert(iterable)
+        elif iterable is not None:
+            raise TypeError('Try again with a list, tuple, int, or float.')
+
+    def insert(self, val):
+        """Insert new node into Binary Search Tree."""
+        if type(val) not in [int, float]:
+            raise TypeError('You can only add numbers to this tree.')
+        curr = self._root
+        iteration = 0
+        right = False
+        rdepth = 0
+        ldepth = 0
+        new_node = False
+        if curr is None:
+            curr = Node(val)
+            self._root = curr
+            self._length = 1
+            self._depth = 1
+            return
+        while True:
+            if val < curr.val:
+                if iteration == 0:
+                        iteration += 1
+                if curr.left:
+                    curr = curr.left
+                else:
+                    curr.left = Node(val)
+                    new_node = True
+            elif val > curr.val:
+                if iteration == 0:
+                        right = True
+                        iteration += 1
+                if curr.right:
+                    curr = curr.right
+                else:
+                    curr.right = Node(val)
+                    new_node = True
+            else:
+                return
+            if right:
+                rdepth += 1
+            else:
+                ldepth += 1
+            if new_node:
+                self._length += 1
+                if ldepth > self._ldepth:
+                    self._ldepth = ldepth
+                if rdepth > self._rdepth:
+                    self._rdepth = rdepth
+                self._depth = max(self._rdepth, self._ldepth) + 1
+                self._balance = self._ldepth - self._rdepth
+                return
+
+    def search(self, val):
+        """Find the node at val in Binary Search Tree."""
+        curr = self._root
+        if type(val) not in [int, float]:
+            raise TypeError('This tree only contains numbers.')
+        while True:
+            if curr is None:
+                return None
```

Comment:
This is unnecessary if your `while` loop depended on the value attached to `curr` instead of simply evaluating `True`